hazelcast/src/main/java/com/hazelcast/jet/impl/connector/WriteMapP.java

```diff
@@ -116,10 +132,14 @@ private boolean submitPending() {
             return false;
         }
         setCallback(map.putAllAsync(buffer));
-        buffer.clear();
+        resetBuffer();
         return true;
     }
 
+    private void resetBuffer() {
+        buffer = new ArrayMap<>(EdgeConfig.DEFAULT_QUEUE_SIZE);
```

Comment:
the buffer is used concurrently only if near-cache is enabled, maybe we can recreate it if near-cache enabled and just clear it if not, wdyt ?

Reply:
Yeah, but that's a current implementation detail. It's not a part of the contract.  In general, if you give data to an async method to deal with, you can't mutate it right after the async method returns.

Reply:
fair enough